### PR TITLE
Migrate southpole modules to Rack v1 api

### PR DESCRIPTION
**Pull request by dogonthehorizon**
This compiles, and I can click around a few things, but I haven't extensively tested patches that use Southpole modules. Where things seemed a little off I've added comments to the files themselves.

Most of the plugins I've been referring to have a `v1` branch that replaces master. Since there isn't a `v1` branch upstream I've just made it against master. I'm happy to change this to a different branch if needed.

Please do let me know if we should change anything else before merging!

### CODE REVIEW

File: plugin.json [hunk 1]
```diff
@@ -0,0 +1,269 @@
+{
+  "slug": "southpole",
+  "name": "southpole",
+  "version": "1.0.0",
+  "license": "MIT",
+  "brand": "southpole",
+  "author": "gbrandt1",
```
Comment (dogonthehorizon) [inline plugin.json:7]: This file needs your attention @gbrandt1!

File: src/Abr.cpp [hunk 2]
```diff
@@ -61,29 +61,32 @@ struct Abr : Module
...
+      reset();
+
+      configParam(Abr::SWITCH1_PARAM, 0.0, 1.0, 0.0, "");
+    }
 
-    void step() override;
+    void process(const ProcessArgs &args) override;
 
-    void reset() override 
+    void reset()
```
Comment (dogonthehorizon) [inline src/Abr.cpp:74]: Several modules overrode `reset()` however it's undetermined if this needs to be changed to `onReset` or not. Same goes for `randomize()` methods.

What stopped me from modifying these methods more intentionally is the fact that some modules depend on being able to call `reset()` in their constructors.

File: src/Aux_.cpp [hunk 3]
```diff
@@ -135,34 +144,30 @@ void Aux::step() {
...
 
 		box.size = Vec(15*4, 380);
 
-		{
-			SVGPanel *panel = new SVGPanel();
-			panel->setBackground(SVG::load(assetPlugin(plugin, "res/Aux_.svg")));
-			panel->box.size = box.size;
-			addChild(panel);	
-		}
+    setPanel(APP->window->loadSvg(asset::plugin(pluginInstance, "res/Aux_.svg")));
```
Comment (dogonthehorizon) [inline src/Aux_.cpp:170]: You can do this in other modules as well. I realized after a while that these were deprecation warnings but weren't harming anything.

File: src/DeuxEtageres.cpp [hunk 4]
```diff
@@ -160,21 +177,22 @@ void DeuxEtageres::step() {
             
             float sumout = lpout*lpgain + hpout*hpgain +  bp2out*bp2gain + bp3out*bp3gain;
 
-            outputs[OUTL_OUTPUT + i].value = sumout;
+            outputs[OUTL_OUTPUT + i].setVoltage(sumout);
     
             // Lights
             lights[CLIPL_LIGHT + i].setBrightnessSmooth( fabs(sumout) > 10. ? 1. : 0. );
```
Comment (dogonthehorizon) [inline src/DeuxEtageres.cpp:183]: This is technically deprecated but it wasn't clear how to migrate to the new method (`setSmoothBrightness` or some such)

File: src/Rakes.cpp [hunk 5]
```diff
@@ -68,21 +67,30 @@ struct Rakes : Module {
...
+      for (int i=0; i < maxsize; i++) {
+        bufl[j][i] = 0;		
+        bufr[j][i] = 0;		
+      }
+      headl[j] = 0;
+      headr[j] = 0;
+    }
+
+    configParam(Rakes::DECAY_PARAM, 0.0, 1.0, 0.0, "");
+//    configParam(Rakes::FOLLOW_PARAM, 0.0, 1.0, 0.0, "");
```
Comment (dogonthehorizon) [inline src/Rakes.cpp:87]: I've left this commented out since the parameter is commented out as well.

File: src/Smoke.cpp [hunk 6]
```diff
@@ -366,185 +378,184 @@ struct SmokeWidget : ModuleWidget {
...
+
+  void step() override {
+    Smoke *smoke = dynamic_cast<Smoke*>(module);
+
+    if (smoke) {
+      panel1->visible = true;
+      panel2->visible = false;
+      panel3->visible = false;
+      panel4->visible = false;
+#ifdef PARASITES
```
Comment (dogonthehorizon) [inline src/Smoke.cpp:528]: A lot of the parasites stuff I generally left alone, it wasn't clear to me if that should be working or not (README seemed to imply that it was working on a branch build?)

**Comment by gbrandt1** [comment]
@dogonthehorizon  thank you very much up to here, i'll take it from here! 
i'll make sure to give you appropriate credit! 
cheers!

**Pull request merged**